blog/2023-11-06-idp-data-preso/index.md

```diff
@@ -0,0 +1,87 @@
+
+## Establishing the source of truth
+
+There are different approaches to representing entities like Kubernetes objects and cloud resources in Backstage. In such context, *platform engineers need to optimize for creation of reliable data*. The last thing you as a platform provider want to see happen is to lose trust of end users because you are displaying incorrect information. There are however, a number of key decisions to be made when building entity representations in Backstage. Particularly:
+
+1. What gets represented in Backstage and what doesnt
+2. How to ensure the Backstage entity offers an accurate representation of its real world counterpart.
+3. What establishes the source of truth for an entity
+
+Embracing GitOps practices, the answer to that last question may sound rather trivial: *GIT, git is obviously the source of truth, since, you know ... GitOps!*
+
+However, while git represents the intended source of truth, truth is actually realized where the resource is deployed, revealing its beloved *resource status*. That is why you may hear people sarcastically refer to git as the *source of hope* in GitOps.
+
+Our current collective of practices reveals that there is no silver bullet when deciding entity representations in a developer portal. What establishes the actual source of truth, from which Backstage entity representations to be drawn, primarily depends on company practices and tools DevOps teams have available to them.
+
+If you operate a Hub and Spoke model, where a single control plane cluster is responsible for handling platform requirements where compute clusters handle the actual workload, the hub cluster could very well become the data source for the Backstage entities. On the other hand, if you operate a federated environment where control plane and data plane workloads are scattered across multiple clusters, Backstage could become the unifier that implements custom entity providers to pull and consolidate data from multiple data sources (i.e. the set of clusters with the right data). In a third approach, the CI may take on the job of hydrating entity definitions with metadata and status information it collects from several data sources, eventually pushing the constructed entity to another repository where it can be observed.
```

Comment:
I think we should also say that a combination of these models could be adopted. For example you could CI to hydrate the catalog file in Git and then use Backstage to finalize the entity based on actualized environment.

Reply:
yeah, added that bit to the conclusion.